test/integration/http_integration.h

```diff
@@ -159,6 +159,11 @@ class HttpIntegrationTest : public BaseIntegrationTest {
   void testDrainClose();
   void testRetry();
   void testRetryHittingBufferLimit();
+  IntegrationStreamDecoderPtr setupPerStreamIdleTimeoutTest();
```

Comment:
Did I miss it or are there test of timeouts on the request path?
I don't think you need to do every stage as it's less interesting than the "have headers been sent" response path but it'd be good to have one.

Reply:
+1, testing the reset case is important.

Reply:
The reset case is being tested, I can add a test for request headers + data sent, then timeout, which is missing. In general, with these tests, its not really implied whether you are timing out on the request or the response path, since the timeout could be due to either party not doing its thang.